src/tree_builder.cpp

```diff
@@ -750,8 +750,8 @@ void HaloBasedTreeBuilder::loop_through_halos(std::vector<HaloPtr> &halos, Simul
 		for(const auto &halo: halos) {
 			halo_snapshots.insert(halo->snapshot);
 		        // compute the minimum particle number for structures
-			for(const auto &subhalo: halo->all_subhalos()) {
-			        if (exec_params.define_transient == ExecutionParameters::ZDEP_3SIGMA || exec_params.define_transient == ExecutionParameters::CONST_10MINPART){
+			if ((exec_params.apply_fix_to_massive_transient_events) && (exec_params.define_transient == ExecutionParameters::ZDEP_3SIGMA || exec_params.define_transient == ExecutionParameters::CONST_10MINPART)){
```

Comment:
The computation of the minimum particle number for a subhalo is only carried out when the fixes for the transients are applied, skipping this step otherwise and making the code run faster.